roles/xsnippet_api/templates/systemd.service.j2

```diff
@@ -2,6 +2,7 @@
 Description = XSnippet API
 After = network.target network-online.target
```

Comment:
> but the same thing could happen with database connection as well.

I'd suggest to add `PostgreSQL` to both `Wants=` and `After=`. So at least after the reboot we can preserve a proper ordering. Or maybe even `Requires=` if we want xsnippet-api not be started if PostgreSQL could not start.

Reply:
Yeah, that sounds good to me! I'm just not sure whether PostgreSQL unit actually waits until the process is ready to accept connections. So we might need infinite (or, at the very least, extra restarts) as defense in depth.

Reply:
If it's linked with systemd, I'd expect it to report status "ready" once it's ready to accept connections.

Reply:
Ah, interesting! Didn't expect that! https://github.com/postgres/postgres/blob/578a7fe7b6f8484f6d7caa2fda288abb3fe87aa0/src/backend/postmaster/postmaster.c#L2317-L2325 confirms there is integration with systemd, indeed.

Let me send add an explicit dependency on `postgresql.service` then.

Reply:
Ah, interesting! Didn't expect that! https://github.com/postgres/postgres/blob/578a7fe7b6f8484f6d7caa2fda288abb3fe87aa0/src/backend/postmaster/postmaster.c#L2317-L2325 confirms there is integration with systemd, indeed.

Let me send add an explicit dependency on `postgresql.service` then.